src/main/java/com/netcracker/study/objects/borders/Wall.java

```diff
@@ -1,22 +1,45 @@
 package com.netcracker.study.objects.borders;
 
-public class Wall implements BorderObject{
-    int thickness;
+public class Wall extends BorderObject{
 
+    private int thickness;
+
+    public Wall() {
+        this.thickness = 1;
+    }
+
+    public Wall(int thickness) {
+        this.thickness = thickness;
+    }
+
+    @Override
     public void onShoot() {
         //Nothing should happen, at least, until messages implementation
     }
 
-    public Wall() {
-        this.thickness = 1;
+    @Override
+    public boolean doesStopBullet(){
```

Comment:
isAlive() ?

Reply:
I've get it... you need to add immutable property for objects implementations and which is going to be checked when onDamage() method is invoked

Reply:
Stopping a bullet does not depend of whether object is "alive" or not. BorderObject might be alive and do not stop bullets (like a windowed wall)